src/main/java/org/cbioportal/proxy/ProxyExchangeGateway.java

```diff
@@ -0,0 +1,197 @@
+import org.springframework.boot.context.properties.ConfigurationPropertiesScan;
+import org.springframework.cloud.gateway.mvc.ProxyExchange;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.StringHttpMessageConverter;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponents;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import jakarta.servlet.http.HttpServletRequest;
+
+@RestController
+@RequestMapping("/proxy")
+@ConfigurationProperties(prefix = "proxy")
+@ConfigurationPropertiesScan
+public class ProxyExchangeGateway {
+	private Map<String, String> routes;
+
+	@Value("${oncokb.token:}")
+	private String oncokbToken;
+
+	@Value("${server.servlet.context-path:}")
+	private String contextPath;
+
+	@Value("${oncokb.public_api.url:https://public.api.oncokb.org/api/v1}")
+	private String oncokbApiUrl;
+
+	@Value("${show.oncokb:false}")
+	private Boolean showOncokb;
+
+	@RequestMapping("/**")
+	public ResponseEntity<?> proxy(ProxyExchange<byte[]> proxy, @RequestBody(required = false) String body,
+			HttpServletRequest request, HttpMethod method, @RequestParam(required = false) MultiValueMap<String, String> params) throws Exception {
+		String service = proxy.path("/proxy/").toString();
+
+		int endPosition = service.indexOf("/");
+		if (endPosition >= 0) {
+			service = service.substring(0, endPosition);
+		}
+
+		if (!routes.containsKey(service)) {
+			throw new UnknownServiceException();
+		}
+
+		String updatedPath = proxy.path("/proxy/" + service).trim();
+
+		String proxyServerHost = routes.get(service).trim();
+
+		ResponseEntity<?> response = null;
+
+		UriComponents uriComponents = UriComponentsBuilder.fromHttpUrl(proxyServerHost + updatedPath).queryParams(params).build();
+
+		switch(method.name()) {
+			case "DELETE":
+				response = proxy.uri(uriComponents.toUri()).body(body).delete();
+				break;
+			case "GET":
+				response = proxy.uri(uriComponents.toUri()).body(body).get();
+				break;
+			case "PATCH":
+				response = proxy.uri(uriComponents.toUri()).body(body).get();
+				break;
+			case "POST":
+				response = proxy.uri(uriComponents.toUri()).body(body).post();
+				break;
+			case "PUT":
+				response = proxy.uri(uriComponents.toUri()).body(body).put();
+				break;
+			default:
+				throw new UnknownServiceException();
+
```

Comment:
I would you HttpMethod.GET and etc instead of raw string.

Also combine GET and PATCH

```java
static boolean itsTheWeekend(DayOfWeek dow) {
    return switch(dow) {
        case SATURDAY, SUNDAY -> true;
        default               -> false;
    };
}
```